builder/amazon/common/step_run_source_instance.go

```diff
@@ -116,6 +117,26 @@ func (s *StepRunSourceInstance) Run(ctx context.Context, state multistep.StateBa
 		EbsOptimized:        &s.EbsOptimized,
 	}
 
+	if s.NoEphemeral {
+		// This is only relevant for windows guests. Ephemeral drives by
+		// default are assigned to drive names xvdca-xvdcz.
+		// When vms are launched from the AWS console, they're automatically
+		// removed from the block devices if the user hasn't said to use them,
+		// but the SDK does not perform this cleanup. The following code just
+		// manually removes the ephemeral drives from the mapping so that they
+		// don't clutter up console views and cause confusion.
+		log.Printf("no_ephemeral was set, so creating drives xvdca-xvdcz as empty mappings")
+		DefaultEphemeralDeviceLetters := "abcdefghijklmnopqrstuvwxyz"
+		for i := 0; i < len(DefaultEphemeralDeviceLetters); i++ {
```

Comment:
Nit: you can also range on the string: https://play.golang.org/p/iQ2DiyRp5MJ

Reply:
hah, I'd just _assumed_ that golang wouldn't make it that easy for me 😂

Reply:
😂 Sometimes it does.

[ also this is sort of complicated in the backscene because c is a rune and then you have to 'cast' it back to a string. Check out, this example makes if fun : https://blog.golang.org/strings#TOC_6.  ]